cmake/thirdparty/FindARPACK.cmake

```diff
@@ -0,0 +1,31 @@
+# Copyright (c) 2019-2024, Lawrence Livermore National Security, LLC and
+# other Serac Project Developers. See the top-level LICENSE file for
+# details.
+#
+# SPDX-License-Identifier: (BSD-3-Clause)
+#
+# Defines the following:
+# ARPACK_FOUND
+# ARPACK_INCLUDE_DIRS
+# ARPACK_LIBRARIES
+
+# Set ARPACK_DIR to specify the prefix of ARPACK installation
+
+if(ARPACK_DIR)
+    set(ARPACK_INCLUDE_DIRS "${ARPACK_DIR}/include")
+
+    foreach(_lib_dir ${ARPACK_DIR}/lib;${ARPACK_DIR}/lib64)
+      if (EXISTS ${_lib_dir})
+        set(ARPACK_LIBRARIES "${_lib_dir}/libparpack.so")
```

Comment:
This line is an issue for MacOS builds -- I need `set(ARPACK_LIBRARIES "${_lib_dir}/libparpack.dylib")`

Reply:
Can we not use the `pkgconfig` files generated by ARPACK?

Reply:
Other than this minor issue, this builds fine locally!

Reply:
Ah, good point.

Reply:
@zatkins-work Let me know if my recent changes resolves your build issues.

Reply:
You should be using `find_library`:

https://cmake.org/cmake/help/latest/command/find_library.html

Reply:
You should also search for a specific header so you know that the installation works.  Here is an example:

https://github.com/LLNL/axom/blob/develop/src/cmake/thirdparty/FindC2C.cmake

Reply:
just fixed that, thanks